### DISCUSSION TITLE
Is this the main benefit after Fast Compiles is enabled?

### DISCUSSION THREAD

**Discussion by dominikj111** [bevyengine/bevy#9397] [Q&A]
I was setting the fast compiles on my MacOS machine.
First impression wasn't really positive as I've spent couple hours to sort it out.

I carried on as written in [Bevy documentation](https://bevyengine.org/learn/book/getting-started/setup/#enable-fast-compiles-optional) and did bash profile tweaks as documented in the `llvm` package installation.

```
export PATH="/usr/local/opt/llvm/bin:$PATH"
export LDFLAGS="-L/usr/local/opt/llvm/lib -L/usr/local/opt/llvm/lib/c++ -Wl,-rpath,/usr/local/opt/llvm/lib/c++"
export CPPFLAGS="-I/usr/local/opt/llvm/include"
```

I summarised the process on [reddit](https://www.reddit.com/r/bevy/comments/15m50gr/do_i_still_need_to_do_anything_special_to_enable/).

On the end I was missing the last step which was to set the `rustflags` in the `config.toml` file. And this step is missing in the documentation and I think it is crucial.

After 20 trials of the `cargo build --target x86_64-apple-darwin` with `rustflags = ["-C", "link-arg=-fuse-ld=lld"]` and without, the average build time with the rustflags was 0.94s and without 1.65s.

I was testing upon small ["hello world"](https://github.com/dominikj111/playground/tree/main/Rust/bevy/hello-2d) project and I think that for bigger project the difference between both times will grow.
Am I right?

Is there another way how to confirm that fast compiles works correctly?

**Comment by nicopap** [comment]
On MacOS, you could also use the [`mold`](https://github.com/rui314/mold) linker, which is significantly faster than `lld`. The documentation is out of date and should be updated, mold works on MacOS. I suspect that now that you figured out the difficult stuff, switching to mold is relatively easy :)

The `config.toml` is mentioned in the page you linked, but only on the section for mold, which doesn't make sense since it's just after a paragraph saying that mold doesn't work with MacOS :zany_face: I suspect it got accidentally moved at one point.

**Reply by dominikj111** [reply]
Thanks, I'll do this during a weekend and will return back :)

**Reply by dominikj111** [reply]
Written on repo you give me ...
> This repository contains a free version of the mold linker. If you are looking for a commercial version that supports macOS please visit the [repository of the sold linker](https://github.com/bluewhalesystems/sold).

So I suppose I'm done here :)
